report/4.Solution/LanuageOfCompiler.tex

```diff
@@ -0,0 +1,30 @@
+\section{The languages of the compiler}
+When creating a compiler there are three languages of relevance, as see in the tombstone diagram\ref{fig:TStoneExample}.
+the input language, in the case of this report HCL, the output language, which is going to be C++. 
+
+\begin{figure}[H]
+	\centering
+	\includegraphics[width=\textwidth/2+\textwidth/4]{4.Solution/images/T-diagram.png}
+	\caption{
+		Example of T-stone diagram showcasing an example of three languages for a given compiler.\cite{TStoneWiki}
+	}
+	\label{fig:TStoneExample}
+\end{figure}
+\subsection{Language of the compiler}
+The compiler was written in Kotlin\cite{KotlinWebsite}, a Java based language.
+This was based on a number of factors.
+The group wanted to try a new language, but didn't have a lot of time, so the language had to look like known paradigms. 
+The group knew C\# and a bit Java, however as the HCL language has a lot of functional features, it was deemed important that the group became comfortable with a language that, at the very least was multi paradigm.
+Kotlin is java-based and is similar, in syntax, to C\# and java, which was important.
+However it is still possible to write methods that is utilizing a functional domain.
+
+This grants the develop af large amount of possibilities, while still being relatively easy to learn.
+A disadvantage could be that low-level languages might have faster executing time, but as HCL is built with the Arduino system in mind, it is not possible to create vast programs, as the architecture has some hardware limitations, so execution time of the compiler will never really affect the end user, as the difference shouldn't be noticeable.
```

Comment:
Maybe change "Arduino system" -> "Arduino architecture", since it is properly defined.

Reply:
Also change "shouldn't" -> "should not"